### ISSUE TITLE
Changes in SSL handling breaks bschmitt/laravel-amqp

### ISSUE THREAD

**Issue by olanko** [php-amqplib/php-amqplib#672]
Hi,

This check in `AMQPSSLConnection` 
```
        if (!isset($ssl_options['verify_peer'])) {
            $ssl_options['verify_peer'] = true;
        }
```

breaks `bschmitt/laravel-amqp` which defines `$ssl_options => []` which means that `ssl` is not enabled and no encryption should be used. Above check ignores this and enforces `ssl` to be used everywhere in `PhpAmqpLib/Wire/IO/StreamIO.php`:
```
        if (!empty($options['ssl'])) {
            if (isset($ssl_protocol)) {
                $this->protocol = $ssl_protocol;
            } else {
                $this->protocol = 'ssl';
            }
        }
```

After setting verify_peer `$options['ssl']` is always non-empty and ssl is forced.

Fix for this would be additional check for `!empty($ssl_options)` in `AMQPSSLConnection`. 

```
        if (!empty($ssl_options) && !isset($ssl_options['verify_peer'])) {
            $ssl_options['verify_peer'] = true;
        }
```

Tested this and it would make `bschmitt/laravel-amqp` work again without extra work.

BR,
-Olli

**Comment by lukebakken** [comment]
We would be happy to review and accept a pull request with this change, thanks.

**Comment by ramunasd** [comment]
@olanko What if we would use $ssl_protocol argument as a switch for secure/non-secure connection? Then $ssl_options content would be ignored if protocol is empty.
In the future I will merge AMQPSSLConnection into same AMQPStreamConnection, because there is no point to keep separate class as a wrapper.

**Comment by olanko** [comment]
@ramunasd That sounds good! Actually we tried yesterday setting $ssl_protocol to null as it would be so obvious :)

**Comment by lukebakken** [comment]
Fixed by #677
